app/services/request_password_reset.rb

```diff
@@ -60,8 +60,10 @@ def user
 
   def email_address_record
     @email_address_record ||= begin
-      EmailAddress.confirmed.find_with_email(email) ||
-        EmailAddress.unconfirmed.find_with_email(email)
+      EmailAddress.transaction do
```

Comment:
This is caused by an error in using `default_scopes` in tests [here](https://github.com/18F/identity-idp/pull/5762/files#diff-675c314b56c4208670e757911eefb5e589c017036ddad2b4a8deab6ac83b76c8L114-L119).  We can ensure consistency by querying within a transaction rather than enforcing `order by` in the test.

Reply:
What if we rewrote this to be one query? Something like `ORDER BY confirmed_at ASC NULL LAST LIMIT 1`

Reply:
I like that a lot!

I think it's probably best handled outside of this PR though, and I'm happy to take that on if that's easier.

Reply:
Sure! Whatever unblocks this best